sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -471,28 +474,96 @@ public <T> CosmosPagedFlux<T> queryItems(String query, Class<T> classType) {
      */
     public Mono<Void> openConnectionsAndInitCaches() {
 
-        if(isInitialized.compareAndSet(false, true)) {
-            return withContext(context -> openConnectionsAndInitCachesInternal()
+        if (isInitialized.compareAndSet(false, true)) {
```

Comment:
we probably can just call openConnectionsAndInitCaches(1)

Reply:
I think I got why we can not do that - because we want to avoid to make it breaking change. As currently, customer can call `openConnectionsAndInitCaches() ` without preferredRegions being configured. So it need to have its own validation. 

Can we add a comments here to mention the above difference?

Reply:
Correct - avoiding the preferred region validation